controllers/kubeobjects/velero/requests.go

```diff
@@ -406,13 +406,38 @@ func getBackupSpecFromObjectsSpec(objectsSpec ramendrv1alpha1.KubeObjectsSpec) v
 		OrLabelSelectors:        objectsSpec.OrLabelSelectors,
 		TTL:                     metav1.Duration{}, // TODO: set default here
 		IncludeClusterResources: objectsSpec.IncludeClusterResources,
-		Hooks:                   velero.BackupHooks{},
+		Hooks:                   getBackupHooks(objectsSpec.KubeResourcesSpec.Hooks),
 		VolumeSnapshotLocations: []string{},
 		DefaultVolumesToRestic:  new(bool),
 		OrderedResources:        map[string]string{},
 	}
 }
 
+func getBackupHooks(hooks []ramendrv1alpha1.HookSpec) velero.BackupHooks {
+	hookSpec := velero.BackupHooks{}
+
+	for _, hook := range hooks {
+		hook := hook // exportloopref: fix variable into local variable
```

Comment:
One way to workaround the second return value of `range` being a copy is to create a reference yourself:
```golang
for i := range hooks {
	hook := &hooks[i]
```